### DISCUSSION TITLE
Tuple to Sequence in strict mode

### DISCUSSION THREAD

**Discussion by ToBeReplaced** [pydantic/pydantic#8786] [Pydantic V2]
I stumbled on: https://github.com/pydantic/pydantic/issues/8173

I understand that per the documentation, `tuple` does not convert to `Sequence` in strict mode.

One reason to want to accept a `tuple` as a `Sequence` is that `tuple` is cheaper to create than `list`, and many parsers (such as https://github.com/msgpack/msgpack-python) will prefer to create tuples instead of lists accordingly.

Moreover, a `tuple` _is_ a `Sequence`, so the rejection is surprising.

What is the rationale for disallowing `tuple`? If this is something that wasn't considered, would the project be welcoming to a pull request that allows `tuple`, or would that be considered a breaking change?

**Comment by sydney-runkle** [comment]
@ToBeReplaced,

Great points. We do think that `tuple` validation should be supported in `strict` mode for `Sequence` validators. I'm a bit surprised that it's not already. I'll create an issue regarding refactoring the `Sequence` validation logic, and include this as one of the acceptance criteria.

**Reply by mwalsh161** [reply]
Did a new ticket ever get opened for this? Running into the same issue.

Thanks!

**Reply by sydney-runkle** [reply]
I think this was related, but I can't seem to find an issue that I opened: https://github.com/pydantic/pydantic/pull/8614

**Reply by sydney-runkle** [reply]
I've opened an issue here: https://github.com/pydantic/pydantic/issues/9340
